src/core/Akka/Actor/ActorPath.cs

```diff
@@ -92,13 +91,42 @@ public override int GetHashCode()
             #endregion
         }
 
+         /** INTERNAL API */
+        internal static char[] ValidSymbols = @"""-_.*$+:@&=,!~';""()".ToCharArray();
+
         /// <summary> 
-        /// The regex that actor names must conform to RFC 2396, http://www.ietf.org/rfc/rfc2396.txt
+        /// Method that checks if actor name conforms to RFC 2396, http://www.ietf.org/rfc/rfc2396.txt
         /// Note that AKKA JVM does not allow parenthesis ( ) but, according to RFC 2396 those are allowed, and 
         /// since we use URL Encode to create valid actor names, we must allow them.
         /// </summary>
-        public static readonly Regex ElementRegex =
-            new Regex(@"^(?:[-a-zA-Z0-9:@&=+,.!~*'_;\(\)]|%[0-9a-fA-F]{2})(?:[-a-zA-Z0-9:@&=+,.!~*'\$_;\(\)]|%[0-9a-fA-F]{2})*$", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        public static bool IsValidPathElement(string s)
+        {
+            return s.Length > 0 && !s.StartsWith("$") && Validate(s.ToCharArray(), s.Length);
```

Comment:
Currently nulls are not going to happen as we generate the name when null is passed, but I agree since it's public. I'll add it.